File: python/svix/models/background_task_type.py
# this file is @generated
from enum import Enum


class BackgroundTaskType(str, Enum):
    ENDPOINT_REPLAY = "endpoint.replay"
    ENDPOINT_RECOVER = "endpoint.recover"
    APPLICATION_STATS = "application.stats"
    MESSAGE_BROADCAST = "message.broadcast"
    SDK_GENERATE = "sdk.generate"
    EVENT_TYPE_AGGREGATE = "event-type.aggregate"
    APPLICATION_PURGE_CONTENT = "application.purge_content"

    def __str__(self) -> str:
        return str(self.value)

File: csharp/Svix/Models/BackgroundTaskType.cs
// this file is @generated
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace Svix.Models
{
    [JsonConverter(typeof(StringEnumConverter))]
    public enum BackgroundTaskType
    {
        [EnumMember(Value = "endpoint.replay")]
        EndpointReplay = 1,

        [EnumMember(Value = "endpoint.recover")]
        EndpointRecover = 2,

        [EnumMember(Value = "application.stats")]
        ApplicationStats = 3,

        [EnumMember(Value = "message.broadcast")]
        MessageBroadcast = 4,

        [EnumMember(Value = "sdk.generate")]
        SdkGenerate = 5,

        [EnumMember(Value = "event-type.aggregate")]
        EventTypeAggregate = 6,

        [EnumMember(Value = "application.purge_content")]
        ApplicationPurgeContent = 7,

        [EnumMember(Value = "endpoint.bulk-replay")]
        EndpointBulkReplay = 8,
    }
}
